monkey/monkey_island/cc/repository/agent_binary_repository.py

```diff
@@ -0,0 +1,29 @@
+import logging
+from typing import BinaryIO
+
+from . import AgentRetrivalError, FileRetrivalError, IAgentBinaryRepository, IFileRepository
+
+logger = logging.getLogger(__name__)
+
+AGENTS = {"linux": "monkey-linux-64", "windows": "monkey-windows-64.exe"}
+
+
+class AgentBinaryRepository(IAgentBinaryRepository):
+    def __init__(self, file_repository: IFileRepository):
+        self._file_repository = file_repository
+
+    def get_linux_binary(self) -> BinaryIO:
+        try:
+            agent_binary = self._file_repository.open_file(AGENTS["linux"])
+            return agent_binary
+        except FileRetrivalError as err:
+            logger.error(f"An error has occured retriving linux agent binary: {err}")
+            raise AgentRetrivalError(f"It couldn't retrieve linux agent binary: {err}")
```

Comment:
Don't log and raise. Just raise. Whatever catches this exception can choose whether or not to log it.

```suggestion
            raise AgentRetrivalError(f"An error occurred while retrieving the Linux agent binary: {err}") from err
```

Reply:
`raise AgentRetrivalError("An error occurred while retrieving the Linux agent binary") from err`

Reply:
@ilija-lazoroski 
![image](https://user-images.githubusercontent.com/19957806/172366556-edd8de2a-7707-4378-a7d9-9f61e2a7084a.png)

@VakarisZ I made the same mistake initially, but you don't need `from err`.

> Exception chaining happens automatically when an [except](https://docs.python.org/3/reference/compound_stmts.html#except)ion is raised inside an except or [finally](https://docs.python.org/3/reference/compound_stmts.html#finally) section.
https://docs.python.org/3/tutorial/errors.html